storage/src/mmr/journaled.rs

```diff
@@ -1,32 +1,54 @@
 //! An MMR backed by a fixed-item-length journal.
+//!
+//! A journal is used to store all unpruned MMR nodes, and a metadata store is used to preserve each
+//! current peak in case they would otherwise have been be pruned.
 
 use crate::journal::{
     fixed::{Config as JConfig, Journal},
     Error as JError,
 };
+use crate::metadata::{Config as MConfig, Metadata};
 use crate::mmr::{
-    iterator::PeakIterator,
+    iterator::{oldest_provable_pos, oldest_required_proof_pos, PeakIterator},
     mem::Mmr as MemMmr,
     verification::{Proof, Storage},
     Error,
 };
+use bytes::Bytes;
 use commonware_cryptography::Hasher;
-use commonware_runtime::{Blob, Metrics, Storage as RStorage};
-use commonware_utils::SizedSerialize;
-use futures::future::try_join_all;
-use tracing::warn;
+use commonware_runtime::{Blob, Clock, Metrics, Storage as RStorage};
+use commonware_utils::array::U64;
+use tracing::{error, warn};
+
+/// Configuration for a journal-backed MMR.
+#[derive(Clone)]
+pub struct Config {
+    /// The prefix to use for the names of the `commonware-runtime::Storage` partitions used for the
+    /// MMR metadata and backing journal.  "_metadata" and "_journal" will be appended to this
+    /// prefix to form the names of the metadata and journal partitions, respectively.
+    pub partition_prefix: String,
```

Comment:
While I originally suggested this approach, I think requiring each partition to be explicitly specified is more sustainable (and will work better when running on cloud storage).

Reply:
done